backends/tc/ebpfCodeGen.h

```diff
@@ -0,0 +1,327 @@
+/*
+Copyright (C) 2023 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions
+and limitations under the License.
+*/
+
+#ifndef BACKENDS_TC_EBPFCODEGEN_H_
+#define BACKENDS_TC_EBPFCODEGEN_H_
+
+#include "backend.h"
+
+namespace TC {
+
+class ConvertToBackendIR;
+
+//  Similar to class PSAEbpfGenerator in backends/ebpf/psa/ebpfPsaGen.h
+
+class PNAEbpfGenerator {
```

Comment:
I was suggesting this class to derive from `PSAEbpfGenerator`. Some methods such as `emitPreamble`, `emitCommonPreamble`, etc. can be directly re-used.

Reply:
There were some functions which needed to be overridden but were not defined as virtual in PSAEbpfGenerator class. Hence replicated similar code and did not derive from PSAEbpfGenerator class.

Reply:
Can you make them virtual in PSAEbpfGenerator?

Reply:
There is a virtual  function emitInitializerSection which is already defined as virtual but not needed in tc as you mentioned in one of your comments. Also  below class members and few other functions though inherited does not right meaning. 
EBPFPipeline *ingress; 
EBPFPipeline *egress;

I think hence it is better to have PNAEbpfGenerator as separate class

Reply:
Maybe it is possible to build a common abstract class for these back ends?

Reply:
Updated the code to have common abstract class